harvester/ckan_utils.py

```diff
@@ -0,0 +1,150 @@
+import re
+
+# all of these are copy/pasted from ckan core
+# https://github.com/ckan/ckan/blob/master/ckan/lib/munge.py
+
+PACKAGE_NAME_MAX_LENGTH = 100
+PACKAGE_NAME_MIN_LENGTH = 2
+
+MAX_TAG_LENGTH = 100
+MIN_TAG_LENGTH = 2
+
+
+def _munge_to_length(string: str, min_length: int, max_length: int) -> str:
+    """Pad/truncates a string"""
+    if len(string) < min_length:
+        string += "_" * (min_length - len(string))
+    if len(string) > max_length:
+        string = string[:max_length]
+    return string
+
+
+def substitute_ascii_equivalents(text_unicode: str) -> str:
+    # Method taken from: http://code.activestate.com/recipes/251871/
+    """
+    This takes a UNICODE string and replaces Latin-1 characters with something
+    equivalent in 7-bit ASCII. It returns a plain ASCII string. This function
+    makes a best effort to convert Latin-1 characters into ASCII equivalents.
+    It does not just strip out the Latin-1 characters. All characters in the
+    standard 7-bit ASCII range are preserved. In the 8th bit range all the
+    Latin-1 accented letters are converted to unaccented equivalents. Most
+    symbol characters are converted to something meaningful. Anything not
+    converted is deleted.
+    """
+    char_mapping = {
+        0xC0: "A",
+        0xC1: "A",
+        0xC2: "A",
+        0xC3: "A",
+        0xC4: "A",
+        0xC5: "A",
+        0xC6: "Ae",
+        0xC7: "C",
+        0xC8: "E",
+        0xC9: "E",
+        0xCA: "E",
+        0xCB: "E",
+        0xCC: "I",
+        0xCD: "I",
+        0xCE: "I",
+        0xCF: "I",
+        0xD0: "Th",
+        0xD1: "N",
+        0xD2: "O",
+        0xD3: "O",
+        0xD4: "O",
+        0xD5: "O",
+        0xD6: "O",
+        0xD8: "O",
+        0xD9: "U",
+        0xDA: "U",
+        0xDB: "U",
+        0xDC: "U",
+        0xDD: "Y",
+        0xDE: "th",
+        0xDF: "ss",
+        0xE0: "a",
+        0xE1: "a",
+        0xE2: "a",
+        0xE3: "a",
+        0xE4: "a",
+        0xE5: "a",
+        0xE6: "ae",
+        0xE7: "c",
+        0xE8: "e",
+        0xE9: "e",
+        0xEA: "e",
+        0xEB: "e",
+        0xEC: "i",
+        0xED: "i",
+        0xEE: "i",
+        0xEF: "i",
+        0xF0: "th",
+        0xF1: "n",
+        0xF2: "o",
+        0xF3: "o",
+        0xF4: "o",
+        0xF5: "o",
+        0xF6: "o",
+        0xF8: "o",
+        0xF9: "u",
+        0xFA: "u",
+        0xFB: "u",
+        0xFC: "u",
+        0xFD: "y",
+        0xFE: "th",
+        0xFF: "y",
+        # 0xa1: '!', 0xa2: '{cent}', 0xa3: '{pound}', 0xa4: '{currency}',
```

Comment:
is there a reason for commenting these out?

Reply:
no, this was a direct copy/paste from the [source](https://github.com/ckan/ckan/blob/f2550929feed5fbd8f155e71d855aa8d4fb1e1b6/ckan/lib/munge.py#L95-L103)